File: src/main/java/com/example/eyeofsauron/service/MarkerService.java
package com.example.eyeofsauron.service;

import com.example.eyeofsauron.entity.Marker;
import com.example.eyeofsauron.repository.MarkerRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class MarkerService {
    private MarkerRepository markerRepository;

    public List<Marker> getAll() { return markerRepository.findAll(); }

    public Optional<Marker> getById(Long id) { return markerRepository.findById(id); }

    public List<Marker> getFreeOrRouteMarkers(Long id) { return markerRepository.findByRouteIdOrFree(id); }

    public void create(Marker marker) { markerRepository.save(marker); }

    public void update(Marker marker) { markerRepository.save(marker); }

    public void delete(Marker marker) { markerRepository.delete(marker); }

    public void deleteById(Long id) { markerRepository.deleteById(id); }
}

File: src/main/kotlin/com/example/eyeofsauron/service/MarkerService.kt
package com.example.eyeofsauron.service

import com.example.eyeofsauron.entity.Marker
import com.example.eyeofsauron.repository.MarkerRepository
import org.springframework.stereotype.Service

/**
 * Сервис для работы с маркерами
 * @author rodkinsi
 */
@Service
class MarkerService(private val repository: MarkerRepository) {
    fun getAll(): List<Marker> = repository.findAll()

    fun getById(id: Long) = repository.findById(id)

    fun getFreeOrRouteMarkers(routeId: Long) = repository.findByRouteIdOrFree(routeId)

    fun getByRfid(rfid: String) = repository.findByRfid(rfid)

    fun create(marker: Marker) = repository.save(marker)

    fun update(marker: Marker) = repository.save(marker)

    fun delete(marker: Marker) = repository.delete(marker)

    fun deleteById(id: Long) = repository.deleteById(id)
}